### PR TITLE
Precommit testcase for pr81872

### PR DESCRIPTION
**Pull request by annamthomas**
Testcase shows miscompile when dropping disjoint flag from disjoint or
during vectorization.

### CODE REVIEW

File: llvm/test/Transforms/LoopVectorize/X86/pr81872.ll [hunk 1]
```diff
@@ -0,0 +1,109 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
+; RUN: opt -S -passes=loop-vectorize < %s | FileCheck %s
+target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
+target triple = "x86_64-unknown-linux-gnu"
+
+@global = external global ptr addrspace(1), align 8
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/X86/pr81872.ll:6]: unused?

File: llvm/test/Transforms/LoopVectorize/X86/pr81872.ll [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+; PR 81872 explains the issue.
+
+; If we vectorize, we have a miscompile where array IV and thereby value stored in (arr[99],
+; arr[98]) is calculated incorrectly since disjoint or was only disjoint because
+; of dominating conditions. Dropping the disjoint to avoid poison still changes
+; the behaviour since now the or is no longer equivalent to the add.
+; Function Attrs: uwtable
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/X86/pr81872.ll:14]: Nit: not needed, replace by `;`

File: llvm/test/Transforms/LoopVectorize/X86/pr81872.ll [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+; CHECK-NEXT:    [[IV_NEXT]] = add nsw i64 [[IV]], -1
+; CHECK-NEXT:    [[ICMP22:%.*]] = icmp eq i64 [[IV_NEXT]], 90
+; CHECK-NEXT:    br i1 [[ICMP22]], label [[BB6]], label [[BB15]], !prof [[PROF6:![0-9]+]], !llvm.loop [[LOOP7:![0-9]+]]
+; CHECK:       bb6:
+; CHECK-NEXT:    ret void
+;
+bb5:
+  br label %bb15
+
+bb15:                                             ; preds = %bb20, %bb8
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/X86/pr81872.ll:69]: nit: would slightly help readability to call this `loop.header` and bb20 -> `loop.latch`

**Review by fhahn — APPROVED** [review]
LGTM, thanks! (Some minor editorial suggestions inline)

**Comment by fhahn** [comment]
Would also be good prefix title with `[LV]` and link the issue URL in the description so it's properly linked in Github

**Comment by annamthomas** [comment]
oh I missed these comments and landed the change. Will address post commit.

**Pull request merged**